### PR TITLE
chore(Carta Giovani Nazionale): [IOACGN-21] Implements error handling on bucket code visualization

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR implements a proper error handling for CGN bucket code visualization

### CODE REVIEW

File: ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { DiscountBucketCode } from "../../../../../definitions/cgn/merchants/DiscountBucketCode";
+
+/**
+ * Type representing service preference successfully loaded
```
Comment (Undermaken) [inline ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts:4]:
```suggestion
 * Type representing service preference successfully loaded
```
maybe is a copy&paste error

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+type DiscountBucketCodeResponseSuccess = {
+  kind: "success";
+  value: DiscountBucketCode;
+};
+
+/**
+ * Type representing service preference successfully loaded
```
Comment (CrisTofani) [inline ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts:12]:
```suggestion
 * Type representing that there's no bucket code available
```

File: ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+type DiscountBucketCodeNotFound = {
+  kind: "notFound";
+};
+
+/**
+ * Type representing service preference successfully loaded
```
Comment (CrisTofani) [inline ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts:19]:
```suggestion
 * Type representing any other error specified by backend response without the need to be handled by the app
```

File: ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts (hunk 1 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/types/DiscountBucketCodeResponse.ts]: yes thx for noticing b199f44 fixes all the comments

**Review by Undermaken — APPROVED** [review]

**Pull request merged**